docs/src/literate-tutorials/linear_elasticity.jl

```diff
@@ -1,3 +1,354 @@
+close!(ch);
+
+# In addition, we will use Neumann boundary conditions on the top surface, where
+# we add a traction vector of the form
+# ```math
+# \boldsymbol{t}_\mathrm{N}(\boldsymbol{x}) = (20e3) x_1 \boldsymbol{e}_2
+# ```
+traction(x) = Vec(0.0, 20e3 * x[1]);
+
+# On the right boundary, we don't do anything, resulting in a zero traction Neumann boundary.
+# In order to assemble the external forces, $f_i^\mathrm{ext}$, we need to iterate over all
+# facets in the relevant facetset. We do this by using the [`FacetIterator`](@ref).
+
+function assemble_external_forces!(f_ext, dh, facetset, facetvalues, prescribed_traction)
+    ## Create a temporary array for the facet's local contributions to the external force vector
+    fe_ext = zeros(getnbasefunctions(facetvalues))
+    for face in FacetIterator(dh, facetset)
+        ## Update the facetvalues to the correct facet number
+        reinit!(facetvalues, face)
+        ## Reset the temporary array for the next facet
+        fill!(fe_ext, 0.0)
+        for qp in 1:getnquadpoints(facetvalues)
+            ## Calculate the global coordinate of the quadrature point.
+            x = spatial_coordinate(facetvalues, qp, getcoordinates(face))
+            tₚ = prescribed_traction(x)
+            ## Get the integration weight for the current quadrature point.
+            dΓ = getdetJdV(facetvalues, qp)
+            for i in 1:getnbasefunctions(facetvalues)
+                Nᵢ = shape_value(facetvalues, qp, i)
+                fe_ext[i] += tₚ ⋅ Nᵢ * dΓ
+            end
+        end
+        ## Add the local contributions to the correct indices in the global external force vector
+        assemble!(f_ext, celldofs(face), fe_ext)
+    end
+    return f_ext
+end
+#md nothing #hide
+
+# ### Material behavior
+# Next, we need to define the material behavior.
+# In this tutorial, we use plane strain linear isotropic elasticity, with Hooke's law as
+# ```math
+# \boldsymbol{\sigma} = 2G \boldsymbol{\varepsilon}^\mathrm{dev} + 3K \boldsymbol{\varepsilon}^\mathrm{vol}
+# ```
+# where $G$ is the shear modulus and $K$ the bulk modulus. The volumetric, $\boldsymbol{\varepsilon}^\mathrm{vol}$,
+# and deviatoric, $\boldsymbol{\varepsilon}^\mathrm{dev}$ strains, are defined as
+# ```math
+# \begin{align*}
+# \boldsymbol{\varepsilon}^\mathrm{vol} &= \frac{\mathrm{tr}(\boldsymbol{\varepsilon})}{3}\boldsymbol{I}, \quad
+# \boldsymbol{\varepsilon}^\mathrm{dev} &= \boldsymbol{\varepsilon} - \boldsymbol{\varepsilon}^\mathrm{vol}
+# \end{align*}
+# ```
+#
+# Starting from Young's modulus, $E$, and Poisson's ratio, $\nu$, the shear and bulk modulus are
+# ```math
+# G = \frac{E}{2(1 + \nu)}, \quad K = \frac{E}{3(1 - 2\nu)}
+# ```
+# Finally, the stiffness tensor can be calculated as the derivative of stress wrt. strain.
+# Due to the linearity, we can calculate this at any point, and choose zero strain.
+Emod = 200e3 # Young's modulus [MPa]
+ν = 0.3      # Poisson's ratio [-]
```

Comment:
Do you agree that we might want to teach users to wrap these into a struct? Or is this too much for this tutorial?

Reply:
In this case I don't see a reason, since we only pass the stiffness tensor to the element routine.